cpp/common/src/codingstandards/cpp/deviations/CodeIdentifierDeviation.qll

```diff
@@ -336,26 +341,60 @@ class CodeIdentifierDeviation extends TCodeIndentifierDeviation {
     )
   }
 
+  predicate hasLocationInfo(
+    string filepath, int suppressedLine, int suppressedColumn, int endline, int endcolumn
+  ) {
+    exists(Comment commentMarker |
+      this = TSingleLineDeviation(_, commentMarker, filepath, suppressedLine) and
+      suppressedColumn = 1 and
+      endline = suppressedLine
+    |
+      if commentMarker instanceof DeviationEndOfLineMarker
+      then endcolumn = commentMarker.(DeviationEndOfLineMarker).getLocation().getEndColumn()
+      else
+        // Find the last column for a location on the next line
+        endcolumn =
+          max(Location l |
```

Comment:
Perhaps the logic here, and the predicates `getLastLineNumber`, `getLastColumnNumber` from `DeviationsSuppression.ql`, should perhaps be a library in cpp/common